extensions/wikia/BannerNotifications/BannerNotificationsController.class.php

```diff
@@ -0,0 +1,292 @@
+<?php
+/**
+ * Handles adding and rendering of confirmations (green stripe below global nav) and notifications (blue bubble boxes just above the fold)
+ *
+ * @author Maciej Brencz
+ */
+
+class BannerNotificationsController extends WikiaController {
+
+	const SESSION_KEY = 'oasisConfirmation';
+
+	// confirmation types
+	const CONFIRMATION_CONFIRM = 1; // Green
+	const CONFIRMATION_NOTIFY = 2; // Blue
+	const CONFIRMATION_ERROR = 3; // Red
+	const CONFIRMATION_WARN = 4; // Yellow
+
+	// HTML of user messages notification (rendered by MW core -- skin variable)
+
+	public function init() {
+		$this->confirmation = null;
+		$this->confirmationClass = null;
+	}
+
+	/**
+	 * Add confirmation message to the user session (so it persists between redirects)
+	 */
+	public static function addConfirmation($message, $type = 1) {
+		wfProfileIn(__METHOD__);
+
+		if ( !empty( $message ) ) {
+			$_SESSION[self::SESSION_KEY] = array(
+				'message' => $message,
+				'type' => $type,
+			);
+
+			wfDebug(__METHOD__ . " - {$message}\n");
+		}
+
+		wfProfileOut(__METHOD__);
+	}
+
+	/**
+	 * Clear confirmation
+	 */
+
+	public static function clearConfirmation() {
+		$_SESSION[self::SESSION_KEY] = null;
+	}
+
+	/**
+	 * Show confirmation stored in user session
+	 */
+	public function executeConfirmation() {
+		wfProfileIn(__METHOD__);
+
+		// call hook to trigger user messages from extensions
+		#wfRunHooks('SkinTemplatePageBeforeUserMsg', array(&$ntl));
+
+		// add testing confirmation
+		// self::addConfirmation('test');
+
+		if (!empty($_SESSION[self::SESSION_KEY])) {
+			$entry = $_SESSION[self::SESSION_KEY];
+
+			$this->confirmation = $entry['message'];
+
+			switch($entry['type']) {
+				case self::CONFIRMATION_NOTIFY:
+					$this->confirmationClass = 'notify';
```

Comment:
what do you think about storing ```$type``` as a string value in session?
We could avoid this mapping then.